### ISSUE TITLE
Error creating blob for some files

### ISSUE THREAD

**Issue by tangrufus** [peter-evans/create-pull-request#3577]
On the same job, some blobs are created for certain files but not all.

```
Create or update the pull request branch
  /usr/bin/git symbolic-ref HEAD --short
  main
  Working base is branch 'main'
  /usr/bin/git checkout --progress -B 6b59eeb1-d933-42d4-a4f2-7b330fabcb49 HEAD --
  M	"tests/.pest/snapshots/E2E/Console/CommandTest/_TypistTech_PhpMatrix_Console_Command__\342\206\222_it_success_with_data_set_____7_3_1____________auto_______full_______7_3_1_________auto____full__.snap"


// ...

Creating tree objects for local commit ce4063d1922ccfd6c2573e7a7139553116777ef6
  Error: Error creating blob for file '"tests/.pest/snapshots/E2E/Console/CommandTest/_TypistTech_PhpMatrix_Console_Command__\342\206\222_it_success_with_data_set_____7_3_1____________auto_______full_______7_3_1_________auto____full__.snap"': ENOENT: no such file or directory, lstat '/home/runner/work/php-matrix/php-matrix/"tests/.pest/snapshots/E2E/Console/CommandTest/

// ...

Creating tree object for blob at 'resources/all-versions.json' with status 'M'
Creating tree object for blob at 'tests/data/releases-8.json' with status 'M'
```

Example: https://github.com/typisttech/php-matrix/actions/runs/12451384164/job/34759263479#step:7:539

On the same repo, another job successfully created the PR with a very similar setup. See: https://github.com/typisttech/php-matrix/actions/runs/12451464051/job/34759436280

Is there anything I can do to debug the issue?

**Comment by peter-evans** [comment]
Hi @tangrufus 

Thank you for opening this issue. I'll take a look at this when I have chance. My guess is that those long, strange looking file names are causing an issue.

**Comment by peter-evans** [comment]
I think I've figured out the issue and I'm making a fix for it. When using `sign-commits`, unicode is not correctly preserved in filepaths.

**Comment by peter-evans** [comment]
Released a fix in `v7.0.6` / `v7`
